### PR TITLE
Split transactions

### PR DESCRIPTION
**Pull request by whoward**
@mclark have a look at this

as part of this i tidied up a lot of the javascript, if you dont get any part feel free to ask

i'm definitely feeling the itch to add a decorator library, draper would probably be the one of choice

I also added some bower assets, you'll have to run `rake bower:install` to get them

### CODE REVIEW

File: app/models/split_expense_transaction.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+class SplitExpenseTransaction < Expense
+  has_many :partitions, class_name: "Expense", foreign_key: "split_transaction_id"
+
+  def splittable?
```
Comment (mclark) [inline app/models/split_expense_transaction.rb]: looks like everything is splittable? is that intended?

Comment (mclark) [inline app/models/split_expense_transaction.rb]: nevermind, missed the transfers below

**Comment by mclark** [comment]
Code review complete, it looks solid.  Good idea of having the casts module, and a very clean implementation.  I haven't run this code however, so your call if you want to just go ahead and merge.

**Comment by whoward** [comment]
thanks @mclark !

**Pull request merged**